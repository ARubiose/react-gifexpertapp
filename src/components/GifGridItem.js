import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ( { id, tittle, url } ) => {
    return (
        <div className='card animate__animated animate__fadeIn'>
            <img src={ url } alt={ tittle } />
            <p>{ tittle }</p>
        </div>
    )
}

GifGridItem.propTypes = {
    tittle: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
