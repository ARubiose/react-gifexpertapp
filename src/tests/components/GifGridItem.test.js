import React from 'react'
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Testing GridGifItem', () => {

    const tittle = 'A tittle';
    const url = 'https://localhost.algo.jpg';
    const wrapper = shallow( <GifGridItem tittle={ tittle } url={ url }/>);

    test('It must show the component correctly', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test( 'It must have a paragraph with the tittle', () => {
        const p = wrapper.find( 'p' );
        expect( p.text().trim() ).toBe( tittle );
        
    });

    test( 'It must have the image equal to the url and alt of the props', () => {
        const img = wrapper.find( 'img' );
        expect( img.props().src ).toBe( url )
        expect(img.props().alt ).toBe( tittle );
    })

    test('should have animate_fadeIn', () => {
        const div = wrapper.find( 'div' );
        const className = div.props().className
        expect( className.includes( 'animate__fadeIn ') ).toBe( true );
    })
    

})