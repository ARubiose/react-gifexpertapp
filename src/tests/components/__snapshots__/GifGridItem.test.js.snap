// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GridGifItem It must show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A tittle"
    src="https://localhost.algo.jpg"
  />
  <p>
    A tittle
  </p>
</div>
`;
